src/util/safe_print.cpp

```diff
@@ -0,0 +1,200 @@
+/*********************                                                        */
+/*! \file safe_print.cpp
+ ** \verbatim
+ ** Top contributors (to current version):
+ **   Andres Noetzli
+ ** This file is part of the CVC4 project.
+ ** Copyright (c) 2009-2016 by the authors listed in the file AUTHORS
+ ** in the top-level source directory) and their institutional affiliations.
+ ** All rights reserved.  See the file COPYING in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** \brief [[ Add one-line brief description here ]]
+ **
+ ** [[ Add lengthier description here ]]
+ ** \todo document this file
+ **/
+
+#include "safe_print.h"
+
+#include <unistd.h>
+
+/* Size of buffers used */
+#define BUFFER_SIZE 20
+
+namespace CVC4 {
+
+template <>
+void safe_print(int fd, const char* msg) {
+  ssize_t ret = write(fd, msg, safe_strlen(msg));
+  (void)ret;
+}
+
+template <>
+void safe_print(int fd, const std::string* msg) {
```

Comment:
A reference would work, too (and might be a bit nicer), but we don't want to copy std::strings because that could lead to dynamic memory allocation.